mlflow/server/js/src/experiment-tracking/components/experiment-page/components/runs/cells/RowActionsHeaderCellRenderer.tsx

```diff
@@ -0,0 +1,69 @@
+import { DropdownMenu, VisibleIcon, VisibleOffIcon } from '@databricks/design-system';
+import { Theme } from '@emotion/react';
+import React from 'react';
+import { FormattedMessage } from 'react-intl';
+import { RUNS_VISIBILITY_MODE } from '../../../utils/experimentPage.common-utils';
+
+/**
+ * A component used to render "eye" icon in the table header used to hide/show all runs
+ */
+export const RowActionsHeaderCellRenderer = React.memo(
+  (props: { onToggleVisibility: (runUuidOrToggle: string) => void }) => (
+    <DropdownMenu.Root modal={false}>
+      <DropdownMenu.Trigger asChild>
+        <button
+          css={styles.actionButton}
+          data-testid='experiment-view-runs-visibility-column-header'
+        >
+          <VisibleIcon />
+        </button>
+      </DropdownMenu.Trigger>
+
+      <DropdownMenu.Content>
+        <DropdownMenu.Item
+          onClick={() => props.onToggleVisibility(RUNS_VISIBILITY_MODE.HIDEALL)}
+          data-testid='experiment-view-runs-visibility-hide-all'
+        >
+          <DropdownMenu.IconWrapper>
+            <VisibleOffIcon />
+          </DropdownMenu.IconWrapper>
+          <FormattedMessage
+            defaultMessage='Hide all runs'
+            description='Menu option for hiding all runs in the experiment view runs compare mode'
+          />
+        </DropdownMenu.Item>
+        <DropdownMenu.Item
+          onClick={() => props.onToggleVisibility(RUNS_VISIBILITY_MODE.SHOWALL)}
+          data-testid='experiment-view-runs-visibility-show-all'
+        >
+          <DropdownMenu.IconWrapper>
+            <VisibleIcon />
+          </DropdownMenu.IconWrapper>
+          <FormattedMessage
+            defaultMessage='Show all runs'
+            description='Menu option for revealing all hidden runs in the experiment view runs compare mode'
+          />
+        </DropdownMenu.Item>
+      </DropdownMenu.Content>
+
+      {/*  */}
+    </DropdownMenu.Root>
+  ),
+);
+
+const styles = {
+  actionButton: (theme: Theme) => ({
+    background: 'transparent',
+    border: 'none',
+    cursor: 'pointer',
+    padding: '8px',
+    // When visibility icon is next to the ag-grid checkbox, remove the bonus padding
+    '.ag-checkbox:not(.ag-hidden) + &': { padding: '0 1px' },
```

Comment:
@hubertzub-db the visibility icon appears to be shifted about 15px to the right in the OSS version:

<img width="1777" alt="Screen Shot 2023-02-19 at 2 47 44 AM" src="https://user-images.githubusercontent.com/39497902/219980812-d4ede283-2980-43dc-93fb-64c96309a7db.png">

I found that adjusting this line to the following brings it back into alignment:

```
'.ag-checkbox:not(.ag-hidden) + &': { margin: '0 -15px' },
```

However, I'm not sure this is the best approach, since it won't sync back nicely to edge. Do you have recommendations for how to resolve this?

Reply:
The issue is related to a exotic edge case related to CSS specificity, here's a slightly better solution that should be nicely mergeable back to edge:
https://github.com/dbczumar/mlflow/pull/64/files

Reply:
Thanks so much, @hubertzub-db !